lib/require-from-worker.js

```diff
@@ -1,43 +1,23 @@
 'use strict';
 
-const debugGenerator = require('heimdalljs-logger');
-const hashForDep = require('hash-for-dep');
 const AstPlugins = require('./ast-plugins');
-
-const _logger = debugGenerator('ember-cli-htmlbars-inline-precompile');
+let astPlugin = {};
 
 module.exports = {
-  build(options) {
-    let pluginInfo = this.setupDependentPlugins(options.parallelConfig);
-
-    return AstPlugins.setup(pluginInfo, {
-      templateCompilerPath: options.templateCompilerPath,
-      useCache: true,
-    });
-  },
-
-  setupDependentPlugins(parallelConfig) {
-    const plugins = [];
-    const cacheKeys = [];
-
-    parallelConfig.forEach((config) => {
-      const plugin = require(config.requireFile);
-      const buildInfo = plugin[config.buildUsing](config.params);
-
-      plugins.push(buildInfo.plugin);
-
-      if (typeof buildInfo.baseDir === 'function') {
-        const pluginHashForDep = hashForDep(buildInfo.baseDir());
-        cacheKeys.push(pluginHashForDep);
-      } else {
-        _logger.debug('ember-cli-htmlbars-inline-precompile is opting out of caching due to an AST plugin that does not provide a caching strategy: `' + buildInfo.name + '`.');
-        cacheKeys.push((new Date()).getTime() + '|' + Math.random());
+  build(options, cacheKey) {
+    // Caching the plugin info so that call to setUp functions will be made once per worker
+    // and not once per module tranformation
+    let plugin = astPlugin[cacheKey];
```

Comment:
Shouldn't this be `let plugin = cacheKey !== undefined && astPlugin[cacheKey]`?

Reply:
Yes it can be. The only reason I didn't do that way is to prevent storing the value in `astPlugin[cacheKey]` when `cacheKey` is `undefined`.